x-pack/plugins/profiling/public/plugin.tsx

```diff
@@ -5,17 +5,66 @@
  * 2.0.
  */
 
-import { AppMountParameters, CoreSetup, CoreStart, Plugin } from '@kbn/core/public';
+import {
+  AppMountParameters,
+  CoreSetup,
+  CoreStart,
+  DEFAULT_APP_CATEGORIES,
+  Plugin,
+} from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import { from, map } from 'rxjs';
+import type { NavigationSection } from '@kbn/observability-plugin/public';
 import { getServices } from './services';
+import type { ProfilingPluginPublicSetupDeps } from './types';
 
-export class ProdfilerPlugin implements Plugin {
-  public setup(core: CoreSetup) {
+export class ProfilingPlugin implements Plugin {
+  public setup(core: CoreSetup, deps: ProfilingPluginPublicSetupDeps) {
     // Register an application into the side navigation menu
+
+    const links = [
+      {
+        id: 'flamegraphs',
```

Comment:
I've just added `flamegraphs` for now, it doesn't really matter because we have no deep-linking at this point, as far as I can tell. I'll address that separately.